src/tests/meson.build

```diff
@@ -0,0 +1,84 @@
+# test suites binaries declaration. These are native tooling
+# that requires libsign_dep to be build natively too, even if the
+# libecc has been configured with a --cross-file argument for
+# cross compilation
```

Comment:
Looking at this... and the fact that you added all the sources to the executables() down below...

It's possible that people doing a cross build want to test that the cross built binaries were built correctly with the foreign toolchain. Meson natively supports this with the exe_wrapper, see https://mesonbuild.com/Cross-compilation.html#defining-the-environment

If an exe_wrapper is not defined in a cross build and some tests need it, IIRC those tests simply do not get run.

Reply:
Thanks for this point ! I fixed the ec_self_test build to let the host target being linked to the usage of cross-file if any and add a basic cross-file exemple in support/meson targetting armv7-m with a exe_wrapper entry.